lib/pageflow/ability_mixin.rb

```diff
@@ -5,38 +5,70 @@ module AbilityMixin
     def pageflow_default_abilities(user)
       return if user.nil?
 
-      can :read, Folder, :id => user.entries.map(&:folder_id)
+      can :view, [Admin::MembersTab, Admin::RevisionsTab]
+      unless user.admin?
+        can [:create, :update, :destroy], Folder do |folder|
+          Policies::FolderPolicy.new(user, folder).manage?
+        end
 
-      can [:read, :use_files], Entry, :id => user.entry_ids
+        can :configure_folder_on, Account do |account|
+          Policies::AccountPolicy.new(user, account).publish?
+        end
```

Comment:
I'd suggest adding alias methods to the policies which reflect the permission we are trying to check. Otherwise we leak implementation details that the permission to publish coincides with the permission to configure folders. If this changes in the future, we only want to have to make changes to the policy and controller tests - not glue code like this one found here.